Cleanup working with tags tutorial (#558)
foundriesio/docs: source/tutorials/working-with-tags/working-with-tags.rst

.. _tutorial-working-with-tags:

Working With Tags
=================

> Your Factory initially has a single branch, ``main``.

Reviewer: Are we at this point yet?

Author: My understanding is we will be with v91.

All Targets will be tagged with the branches they are built from.

> This means the Targets have the tag ``main``.

Reviewer 2: This isn't exactly true. Your target gets a tag that's set by our tagging rules. By convention, people usually have branch-name == tag-name.

Author: Thanks! I will update!

> All Targets will be tagged with the branches they are built from.
> This means the Targets have the tag ``main``.

Author: ```suggestion
By convention, Targets are tagged with the branches they are built from.
This means the Targets have the tag ``main``.
```

.. warning::
   For this tutorial, it is assumed you have created and worked on a ``devel`` branch.
   This has not yet been covered.
   See the :ref:`ref-factory-sources` reference manual on how to configure the CI with new branches.

This helps keep the development flow fast.
For example, you have  ``platform-devel``, a ``platform`` build based on ``devel`` branch, and install it on the device.

Then you develop applications on ``containers.git`` from the ``devel`` branch.
The  application is built by the CI with a ``containers-devel`` trigger name and produces a Target tagged with ``devel``.

Finally, the device automatically updates to the latest Target tagged with ``devel``.

There are some use cases that you might want to control what tag a device follows, and Targets are tagged:

> - Preventing a device following a tag such as ``devel``, which is automatically created every time you change the ``devel`` branch.

Reviewer: ```suggestion
- Preventing a device from following a tag such as ``devel``, which is automatically created every time you change the ``devel`` branch.
```
